src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -320,11 +333,17 @@ public function listAction(
             'is_shop_context' => $this->get('prestashop.adapter.shop.context')->isShopContext(),
         );
         if ($view !== 'full') {
-            return $this->render('@Product/CatalogPage/Lists/list_' . $view . '.html.twig', array_merge($vars, [
-                'limit' => $limit,
-                'offset' => $offset,
-                'total' => $totalCount,
-            ]));
+            return $this->render(
+                '@Product/CatalogPage/Lists/list_' . $view . '.html.twig',
+                array_merge(
+                    $vars,
+                    [
+                        'limit' => $limit,
+                        'offset' => $offset,
+                        'total' => $totalCount,
+                    ]
+                )
+            );
         }
 
         return $vars;
```

Comment:
this is not clear at all, sometimes we return a response, sometime it's an array..
we should always return the same thing, the convention chosen for new migrated controllers is to always explicitly return a Response via render
could you please remove the `@Template` association and use render instead?
it may not be necessary on all actions (although it would be nice), but this one especially is really messy

Reply:
I just add line break here :P It's from my cleanup commit.

Reply:
I know you didn't implement it, I was just thinking it could be the occasion to improve it
But it can be done later

Reply:
Too risky to change something and or refactor everything before the release :(

Reply:
But I thought about it ^^

Reply:
you're right!